addons/controllers/csi/vspherecsiconfig_utils.go

```diff
@@ -0,0 +1,259 @@
+	}
+
+	// populated from default value in https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/vsphere-csi/2.4.1/bundle/config/values.yaml
+	dvs.VSphereCSI.UseTopologyCategories = false
+	dvs.VSphereCSI.ProvisionTimeout = "300s"
+	dvs.VSphereCSI.AttachTimeout = "300s"
+	dvs.VSphereCSI.ResizerTimeout = "300s"
+	dvs.VSphereCSI.DeploymentReplicas = 3
+	dvs.VSphereCSI.InsecureFlag = true
+
+	// override derived values IF set in csi configuration by user
+	if config.Namespace != "" {
+		dvs.VSphereCSI.Namespace = config.Namespace
+	}
+	if config.ClusterName != "" {
+		dvs.VSphereCSI.ClusterName = config.ClusterName
+	}
+	if config.Server != "" {
+		dvs.VSphereCSI.Server = config.Server
+	}
+	if config.Datacenter != "" {
+		dvs.VSphereCSI.Datacenter = config.Datacenter
+	}
+	if config.PublicNetwork != "" {
+		dvs.VSphereCSI.PublicNetwork = config.PublicNetwork
+	}
+	if config.Username != "" {
+		dvs.VSphereCSI.Username = config.Username
+	}
+	if config.Password != "" {
+		dvs.VSphereCSI.Password = config.Password
+	}
+	if config.Region != "" {
+		dvs.VSphereCSI.Region = config.Region
+	}
+	if config.Zone != "" {
+		dvs.VSphereCSI.Zone = config.Zone
+	}
+	if config.VSphereVersion != "" {
+		dvs.VSphereCSI.VSphereVersion = config.VSphereVersion
+	}
+	if config.WindowsSupport != nil {
+		dvs.VSphereCSI.WindowsSupport = *config.WindowsSupport
+	}
+	if config.TLSThumbprint != "" {
+		dvs.VSphereCSI.TLSThumbprint = config.TLSThumbprint
+	}
+	if config.HttpProxy != "" {
+		dvs.VSphereCSI.HttpProxy = config.HttpProxy
+	}
+	if config.HttpsProxy != "" {
+		dvs.VSphereCSI.HttpsProxy = config.HttpsProxy
+	}
+	if config.NoProxy != "" {
+		dvs.VSphereCSI.NoProxy = config.NoProxy
+	}
+	if config.UseTopologyCategories != nil {
+		dvs.VSphereCSI.UseTopologyCategories = *config.UseTopologyCategories
+	}
+	if config.ProvisionTimeout != "" {
+		dvs.VSphereCSI.ProvisionTimeout = config.ProvisionTimeout
+	}
+	if config.AttachTimeout != "" {
+		dvs.VSphereCSI.AttachTimeout = config.AttachTimeout
+	}
+	if config.ResizerTimeout != "" {
+		dvs.VSphereCSI.ResizerTimeout = config.ResizerTimeout
+	}
+	if config.DeploymentReplicas != nil {
+		dvs.VSphereCSI.DeploymentReplicas = *config.DeploymentReplicas
+	}
+	if config.InsecureFlag != nil {
+		dvs.VSphereCSI.InsecureFlag = *config.InsecureFlag
+	}
+
+	return dvs, nil
+}
+
+// getOwnerCluster verifies that the VSphereCSIConfig has a cluster as its owner reference,
+// and returns the cluster. It tries to read the cluster name from the VSphereCSIConfig's owner reference objects.
+// If not there, we assume the owner cluster and VSphereCSIConfig always has the same name.
+func (r *VSphereCSIConfigReconciler) getOwnerCluster(ctx context.Context,
+	vcsiConfig *csiv1alpha1.VSphereCSIConfig) (*clusterapiv1beta1.Cluster, error) {
+
+	logger := log.FromContext(ctx)
+	cluster := &clusterapiv1beta1.Cluster{}
+	clusterName := vcsiConfig.Name
```

Comment:
nit: could you help add some comments on why vcsiConfig.Name is the clusterName ?

Reply:
comment added